### ISSUE TITLE
ApplicationCommandOptionValueString sets up integer argument

### ISSUE THREAD

**Issue by Annwan** [aquarial/discord-haskell#97]
When I use `ApplicationCommandOptionValueString` to set up a String option to my application command, discord says it requires an integer argument

```hs
 ApplicationCommandOptionValueString
        "group"
        "Group to get the planning for"
        (Just True)
        Nothing
        Nothing
 ```
![image](https://user-images.githubusercontent.com/30665559/152688046-058eafac-e796-46cd-9d60-7bdec534c0cf.png)

I think the problem is here: https://github.com/aquarial/discord-haskell/blob/b14ee1843ec1be5ba5cbfac6e1c2302d9284d50c/src/Discord/Internal/Types/ApplicationCommands.hs#L153 as it uses `ApplicationCommandOptionTypeInteger` in the wrapper, though I may be wrong

**Comment by aquarial** [comment]
Should be fixed on master branch. Will close this once hackage 1.12.0 is released

**Comment by aquarial** [comment]
Should be fixed on 1.12.0

Re-open if there's still an issue
